packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -1302,6 +1302,73 @@ void main() {
     expect(find.text('Cut'), findsNothing);
   });
 
+  testWidgets('Handles the read-only flag correctly', (WidgetTester tester) async {
+    final TextEditingController controller =
+        TextEditingController(text: 'Lorem ipsum dolor sit amet');
+    await tester.pumpWidget(
+      MaterialApp(
+        home: EditableText(
+          controller: controller,
+          backgroundCursorColor: Colors.grey,
+          focusNode: focusNode,
+          style: textStyle,
+          cursorColor: cursorColor,
+        ),
+      ),
+    );
+
+    final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+
+    // Select the first word "Lorem".
+    state.renderEditable.selectWordsInRange(
+      from: Offset.zero,
+      cause: SelectionChangedCause.tap,
+    );
+
+    if (kIsWeb) {
+      // On the web, a regular connection to the platform should've been made
+      // with the `readOnly` flag set to true.
+      expect(tester.testTextInput.hasAnyClients, isTrue);
+      expect(tester.testTextInput.setClientArgs['readOnly'], isTrue);
+      expect(tester.testTextInput.editingState['text'], equals('Lorem'));
+    } else {
+      // On non-web platforms, a read-only field doesn't need a connection with
+      // the platform.
+      expect(tester.testTextInput.hasAnyClients, isFalse);
+    }
+  });
+
+  testWidgets('Does not accept updates when read-only', (WidgetTester tester) async {
+    final TextEditingController controller =
+        TextEditingController(text: 'Lorem ipsum dolor sit amet');
+    await tester.pumpWidget(
+      MaterialApp(
+        home: EditableText(
+          controller: controller,
+          backgroundCursorColor: Colors.grey,
+          focusNode: focusNode,
+          style: textStyle,
+          cursorColor: cursorColor,
+        ),
+      ),
+    );
+
+    final EditableTextState state =
+        tester.state<EditableTextState>(find.byType(EditableText));
+
+    // Select something.
+    state.renderEditable.selectWordsInRange(
+      from: Offset.zero,
+      cause: SelectionChangedCause.tap,
+    );
+
+    expect(tester.testTextInput.hasAnyClients, isTrue);
+    tester.testTextInput.enterText('Foo bar');
+    // No change.
+    expect(controller.text, 'Lorem ipsum dolor sit amet');
+  }, skip: !kIsWeb);
```

Comment:
We treat skips as technical debt. Can the test body be changed so that we simply expect different behavior on different platforms? If not, let's leave a comment explaining why this test is web-only.

/cc @Piinks for more guidance.

Reply:
True! If the test can be expected to work across platforms that would be ideal.
If it's skipped because of an outstanding issue, there should be a comment with the issue link beside it.
If it is being skipped because of platform specific behavior, you could execute the test conditionally based on the platform, or leave a comment saying why the test is skipped without plans to resolve it.